app/src/org/commcare/utils/FileUtil.java

```diff
@@ -455,6 +458,22 @@ public static String getExtension(String filePath) {
         return "";
     }
 
+    /**
+     * Retrieve a file's name using contentUri.
+     * https://developer.android.com/training/secure-file-sharing/retrieve-info.html#RetrieveFileInfo
+     */
+    public static String getFileName(ContentResolver contentResolver, Uri uri) {
+        Cursor cursor = contentResolver.query(uri, null, null, null, null);
+        if (cursor == null) {
+            return "";
+        }
+        int nameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+        cursor.moveToFirst();

```

Comment:
This would crash if there is 0 row returned by the cursor.

Reply:
ohh, thanks for pointing.